src/hooks/useStopPatchPolling.ts

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from "react";
+import { PatchStatus } from "types/patch";
+
+interface Props {
+  hasData: boolean;
+  status: string;
+  activated: boolean;
+  stopPolling: () => void;
+}
+
+export const useStopPatchPolling = ({
+  hasData,
+  status,
+  activated,
+  stopPolling,
+}: Props) => {
+  if (
+    hasData &&
+    (status === PatchStatus.Failed ||
+      status === PatchStatus.Success ||
+      activated === false)
+  ) {
+    stopPolling();
+  }
+};
```

Comment:
perfect! i was thinking that as a precaution against memory leaks that we could add this to the mix: 
```
useEffect(() => stopPolling, [])
```
This way, when the component unmounts, `stopPolling` will always be called